projects/smoke_mcp2515/src/main.c

```diff
@@ -62,15 +99,43 @@ static void prv_periodic_send(SoftTimerId timer_id, void *context) {
   soft_timer_start_millis(500, prv_periodic_send, NULL, NULL);
 }
 
+void prv_send_messages() {
+  LOG_DEBUG("Testing send standard id\n");
+  mcp2515_tx(&s_mcp2515, 0x246, false, 0x1122334455667788, 8);
+  delay_ms(50);
+
+  if ((s_id_loopback == 0x246) && (s_extended == false) && (s_data == 0x1122334455667788) &&
+      (s_dlc == 8)) {
+    LOG_DEBUG("Standard Message Received\n");
+  } else {
+    LOG_DEBUG("Standard message not received\n");
+  }
+
+  LOG_DEBUG("Testing send extended id\n");
+  mcp2515_tx(&s_mcp2515, 0x19999999, true, 0xBEEFDEADBEEFDEAD, 8);
+  delay_ms(50);
+
+  if ((s_id_loopback == 0x19999999) && (s_extended == true) && (s_data == 0xBEEFDEADBEEFDEAD) &&
+      (s_dlc == 8)) {
+    LOG_DEBUG("Extended Message Received\n");
+  } else {
+    LOG_DEBUG("Extended message not received\n");
+  }
+}
```

Comment:
Please add a broad "all loopback checks ok" kind of message to indicate whether or not the loopback was successful. With just this, it isn't 100% obvious whether or not things are working unless you look through the code.